proxy/certs/certs.go

```diff
@@ -215,9 +234,44 @@ func (s *RemoteCertSource) Remote(instance string) (cert *x509.Certificate, addr
 		logging.Errorf("%v", err)
 		logging.Errorf("WARNING: specifying the correct region in an instance string will become required in a future version!")
 	}
-	if len(data.IpAddresses) == 0 || data.IpAddresses[0].IpAddress == "" {
+
+	if len(data.IpAddresses) == 0 {
 		return nil, "", "", fmt.Errorf("no IP address found for %v", instance)
 	}
+
+	// Find the first matching IP address type between user input types and the types that the instance has
+	IPAddrInUse := ""
+	isFound := false
+	for _, eachIPAddrTypeByUser := range s.IPAddrTypes {
+		for _, eachIPAddrTypeOfInstance := range data.IpAddresses {
+			if strings.ToUpper(eachIPAddrTypeOfInstance.Type) == strings.ToUpper(eachIPAddrTypeByUser) {
+				IPAddrInUse = eachIPAddrTypeOfInstance.IpAddress
+				isFound = true
+				break
```

Comment:
We usually prefer to return early, this will skip all the breaks and conditionals and remove the need for isFound

Reply:
Done.